src/Microsoft.AspNet.SignalR.Core/TaskAsyncHelper.cs

```diff
@@ -1354,5 +1354,24 @@ private static class TaskCache<T>
         {
             public static Task<T> Empty = MakeTask<T>(default(T));
         }
+
+        [SuppressMessage("Microsoft.Performance", "CA1811:AvoidUncalledPrivateCode", Justification = "This is a shared file")]
+        public static void Dispatch(Action action)
```

Comment:
A future version could avoid the closure in some cases by supporting a state parameter and `Action<T>`.